Microsoft.Toolkit.Uwp.UI.Controls.Input/RichSuggestBox/RichSuggestBox.cs

```diff
@@ -0,0 +1,943 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Toolkit.Uwp.Deferred;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.Foundation;
+using Windows.Foundation.Metadata;
+using Windows.System;
+using Windows.UI.Input;
+using Windows.UI.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Input;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The RichSuggestBox control extends <see cref="RichEditBox"/> control that suggests and embeds custom data in a rich document.
+    /// </summary>
+    [TemplatePart(Name = PartRichEditBox, Type = typeof(RichEditBox))]
+    [TemplatePart(Name = PartSuggestionsPopup, Type = typeof(Popup))]
+    [TemplatePart(Name = PartSuggestionsList, Type = typeof(ListViewBase))]
+    [TemplatePart(Name = PartSuggestionsContainer, Type = typeof(Border))]
+    [TemplatePart(Name = PartHeaderContentPresenter, Type = typeof(ContentPresenter))]
+    [TemplatePart(Name = PartDescriptionPresenter, Type = typeof(ContentPresenter))]
+    public partial class RichSuggestBox : ItemsControl
```

Comment:
For TTB we split out some of the logic into different partial classes. There's a lot of stuff going on in this one class, are there some things that would make sense to group together across partial class files for organization? (e.g. selection, tokenization, or ui/flyout specific stuff?)

Reply:
I will play around with ways to split them into more files as the main `RichSuggestBox.cs` is rather large at the moment. In the meantime, I will resolve other problems first so they don't get "outdated" and save this organization task for last.

Reply:
Yeah, we could do it as a separate just refactor PR later so we can stay in-sync with suggestions from the PR, good idea.